File: python/BS/BinarySearch.py
class Solution:
    
    def binarySearch(self,nums,target,start,end):
        if start > end:
            return -1
        mid = (start + end) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] > target:
            return self.binarySearch(nums,target,start,mid-1)
        else:
            return self.binarySearch(nums,target,mid+1,end)
    
    def search(self, nums, target):
        # start = 0
        # end = len(nums) - 1
        # while start <= end:
        #     mid = (start + end) // 2
        #     if nums[mid] == target:
        #         return mid
        #     elif nums[mid] > target:
        #         end = mid - 1
        #     else:
        #         start = mid + 1
        # return -1
        return self.binarySearch(nums,target,0,len(nums)-1)
        

n=int(input())
nums=list(map(int,input().split()))
target=int(input())
object=Solution()
print(object.search(nums,target))

File: java/BS/BinarySearch.java
package BS;
import java.util.*;
class BinarySearch
{
    public static int binarySearch(int[] arr,int n,int target)
    {
        int l=0;
        int r=n-1;
        while(l<=r)
        {
            int mid = (l+r)/2;
            if(arr[mid]==target)
            {
                return mid;
            }
            else if(arr[mid]<target)
            {
                l=mid+1;
            }
            else
            {
                r=mid-1;
            }
        }
        return -1;
    }

    public static void main(String[] args)
    {
        try(Scanner sc = new Scanner(System.in)){
        int n=sc.nextInt();
        int[] arr = new int[n];
        for(int i=0;i<n;i++)
        {
            arr[i]=sc.nextInt();
        }
        int target = sc.nextInt();
        int ans = binarySearch(arr,n,target);
        System.out.println(ans);
        sc.close();
    }
    }
}